common/types/nipost.go

```diff
@@ -1,19 +1,22 @@
 package types
 
-//go:generate scalegen -types NIPostBuilderState
+//go:generate scalegen -types NIPostBuilderState,PoetRequest
 
 // NIPostBuilderState is a builder state.
+type PoetRequest struct {
```

Comment:
nit: why not `PoetService`? it's what you use for variable name in the code.

Reply:
Because it describes an in-flight request to a PoetService, not a service itself. It contains the round ID of the submitted challenge. It could be named `SubmittedPoetChallenge`, or something like that.